early-attack-detector-py/src/agent.py

```diff
@@ -262,7 +283,9 @@ def detect_malicious_contract(
     findings = []
 
     if created_contract_address is not None and code is not None:
-        if len(code) > BYTE_CODE_LENGTH_THRESHOLD:
+        from_eoa_in_tp_list = from_ in TP_ATTACKER_LIST
+
+        if (len(code) > BYTE_CODE_LENGTH_THRESHOLD) or from_eoa_in_tp_list:
```

Comment:
If tx.from_ is in TP_ATTACKER_LIST, we are always raising an alert, correct?  Independently of the performance of the model. And we modify the confidence to 1. Then, wouldn't it be better to not modify detect_malicious_contract, and just put the condition
```from_eoa_in_tp_list = from_ in TP_ATTACKER_LIST```
outside of detect_malicious_contract? Or alternatively, skip the model inference, calculation of opcodes, etc. as we won't use it anymore in that case

Reply:
I did initially look at having the check happen earlier (specifically not having to invoke `exec_model()`), but I saw that the findings include the `opcode_addresses` returned from `exec_model()`. So I tried to incorporate the TP list without changing too much since we weren't able to get your input.

However, it does seem like you are onboard with proceeding with refactoring it without having some of the `metadata` in the finding if the contract creator is in the TP list. Any other details I should keep in mind while refactoring?

Reply:
Ultimately, it sort of comes down to which info should we provide in the alert's `metadata` in the case when `transaction_event.from_` is in the `TP_ATTACKER_LIST`. Because I could adjust the logic to have the check happen sooner and exclude unnecessary steps to generate the `metadata` we decide we don't need.

Reply:
Fair point, I didn't think of that. Then I'm ok as it is, `predict_proba` is very inexpensive so we can run it anyway